src/types.ts

```diff
@@ -261,3 +282,206 @@ export interface CreatedComment {
 export const CreatedCommentSpec: ObjectSpec<CreatedComment> = {
   id: int,
 };
+
+export interface CreateBugContent {
+  product: string;
+  component: string;
+  summary: string;
+  version: string;
+  description: string;
+  op_sys: string;
+  platform: string;
+  priority: string;
+  severity: string;
+  alias?: string[];
+  assigned_to?: string;
+  cc?: string[];
+  comment_is_private?: boolean;
+  comment_tags?: string[];
+  groups?: string[];
+  keywords?: string[];
+  qa_contact?: string;
+  status?: string;
+  resolution?: string;
+  target_milestone?: string;
+  flags?: SetFlag[];
+}
+
+export const CreateBugContentSpec: ObjectSpec<CreateBugContent> = {
+  product: string,
+  component: string,
+  summary: string,
+  version: string,
+  description: string,
+  op_sys: string,
+  platform: string,
+  priority: string,
+  severity: string,
+  alias: optional(array(string)),
+  assigned_to: optional(string),
+  cc: optional(array(string)),
+  comment_is_private: optional(boolean),
+  comment_tags: optional(array(string)),
+  groups: optional(array(string)),
+  keywords: optional(array(string)),
+  qa_contact: optional(string),
+  status: optional(string),
+  resolution: optional(string),
+  target_milestone: optional(string),
+  flags: optional(array(object(SetFlagSpec))),
+};
+
+export interface CreatedBug {
+  id: int;
+}
+
+export const CreatedBugSpec: ObjectSpec<CreatedBug> = {
+  id: int,
+};
+export interface UpdateTemplate<T> {
+  add?: T[];
+  remove?: T[];
+}
+
+export const UpdateTemplateStringSpec: ObjectSpec<UpdateTemplate<string>> = {
+  add: optional(array(string)),
+  remove: optional(array(string)),
+};
+
+export const UpdateTemplateIntSpec: ObjectSpec<UpdateTemplate<int>> = {
+  add: optional(array(int)),
+  remove: optional(array(int)),
+};
+
+export interface UpdateTemplateWithOverwrite<T> extends UpdateTemplate<T> {
+  set?: T[];
+}
```

Comment:
```suggestion
export type UpdateOrReplaceList<T> = UpdateList<T> | {
  set: T[];
}
```

Given the docs it looks like the API ignores add and remove when set is present so I think this type makes it clearer what is desired.

Reply:
You are correct, thanks!